### PR TITLE
Tasks without assignees

### PR DESCRIPTION
**Pull request by nickoferrall**
PR for issue #4115 

### Consideration

- The `/team` and `/me/tasks` pages do not show unassigned tasks. However, if a user unassigns a task, they will see the unassigned task (a task with a question mark instead of an avatar) until they refresh the page. I think this is preferable compared with immediately removing the task from `/team` or `/me/tasks` as a user may accidentally unassign a task and want to reassign it.

### AC

- [ ]  the Task object is refactored to support null userId
- [ ]  all task mutations are updated to support a Task with a nullable assignee (e.g. createTask, updateTask, deleteTask, etc etc)
- [ ]  A Task card shows a question mark as the assignee if it doesn't have an assignee
- [ ]  The Task assignee dropdown in the Task card footer includes the current assignee's name. if already selected & clicked again, the task becomes unassigned
- [ ]  the tasks query returns only tasks that have assignees

### CODE REVIEW

File: packages/server/graphql/types/Team.ts
Comment (nickoferrall) [inline packages/server/graphql/types/Team.ts:231]: We could use the tasks query here instead of the resolve function. As both could work, I thought I'd keep things as they are and filter from here.

File: packages/client/modules/outcomeCard/components/OutcomeCardAssignMenu/TaskFooterTeamAssigneeMenu.tsx
Comment (nickoferrall) [inline packages/client/modules/outcomeCard/components/OutcomeCardAssignMenu/TaskFooterTeamAssigneeMenu.tsx:42]: Instead of filtering `teams`, I thought we could include the team the task is currently assigned to. In production it currently shows "Move to" menu with no items in the menu if there are no other teams.

This PR shows this currently assigned team. But, unlike the `/team` page, clicking the currently assigned team doesn't unassign it; the update function doesn't fire.

File: packages/client/modules/outcomeCard/components/OutcomeCardAssignMenu/TaskFooterUserAssigneeMenu.tsx
Comment (nickoferrall) [inline packages/client/modules/outcomeCard/components/OutcomeCardAssignMenu/TaskFooterUserAssigneeMenu.tsx:28]: TypeScript complains if we try to map through `teamMembers` as it says it's readOnly. I tried to cast it to an interface but it wouldn't accept that. I tried a few solutions from [here](https://stackoverflow.com/questions/46634876/how-can-i-change-a-readonly-property-in-typescript) such as `type Writeable<T> = {-readonly [P in keyof T]-?: T[P]}` with no joy.

I'm reluctantly using an `any` at the moment, (TypeScript was previously recognising `assignees` as `any` too) but I'd love to know if there's a better way

File: packages/client/mutations/CreateJiraIssueMutation.ts
Comment (nickoferrall) [inline packages/client/mutations/CreateJiraIssueMutation.ts:64]: It looks Jira integrations require a `user` so I'm removing the integration button from the unassigned tasks

**Comment by mattkrick** [comment]
pinging @tianrunhe for a review!

**Comment by tianrunhe** [comment]
Apologize for not responding! I checked this last week and thought it had a couple of conflicts (from the files and also functionality wise) with #4105. I feel like we should merge my changes in first and revisit this one. @nickoferrall if you need this sooner, definitely let me know and we can work together to resolve the conflicts.

**Comment by nickoferrall** [comment]
No problem, I'm happy to wait until your PR has been merged

**Comment by mattkrick** [comment]
@nickoferrall looks like a couple merge conflicts, other than that g2g!

File: packages/client/modules/outcomeCard/components/OutcomeCardAssignMenu/TaskFooterTeamAssigneeMenu.tsx
Comment (nickoferrall) [inline packages/client/modules/outcomeCard/components/OutcomeCardAssignMenu/TaskFooterTeamAssigneeMenu.tsx:32]: Also here @tianrunhe & @mattkrick , by filtering the currently assigned team from the dropdown, we get an empty menu if the user is only in one team:

empty-menu

We have an issue open for this [here](https://github.com/ParabolInc/parabol/issues/4129). In this file, I'm including the `teamId` in the menu so it always shows something. If the user clicks the currently assigned team, the mutation doesn't fire.

**Review by nickoferrall — COMMENTED** [review]
Merge conflicts fixed (although it looks like there's a problem with the typechecks in `master`?). I've added two comments about updates involving the recent user task filter changes

**Comment by tianrunhe** [comment]
@nickoferrall Does this PR need update against the latest master branch?

**Comment by nickoferrall** [comment]
yep - thanks @tianrunhe, just fixed the merged conflicts

**Comment by tianrunhe** [comment]
I find the un-assignment not that intuitive for our user to figure it out. Maybe a ❌ before the current task owner name? @ackernaut what do you think?

**Comment by ackernaut** [comment]
@tianrunhe I think that’s okay. We don’t really want many unassigned tasks. Each item can be selected or not. What about [this look for that avatar and label](https://www.figma.com/file/AcSOOEtScsOqcOB4iqowMZ/Unassigned?node-id=1%3A10)?

**Review by tianrunhe — CHANGES_REQUESTED** [review]
@nickoferrall  If you could update to the icon Terry provided, that'd be super cool. Otherwise the code LGTM!

**Comment by nickoferrall** [comment]
@tianrunhe sure thing, I like the look of that unassigned icon. Just pushed the update

**Comment by ackernaut** [comment]
@nickoferrall I like the idea of using the gray text color for the ‘Unassigned’ label. What do you think?

**Comment by nickoferrall** [comment]
@ackernaut I like it - it helps to differentiate the label from the input text. I've pushed an update using the gray label and `PALETTE.TEXT_MAIN` on hover to be consistent with the menu in `TaskFooterTeamAssignee`

**Comment by ackernaut** [comment]
`TEXT_MAIN_HOVER` is slightly darker than `TEXT_MAIN` but I’m okay with the change. Thanks for making that change, this polish is nice :)

**Comment by ackernaut** [comment]
Nah, `TEXT_MAIN_HOVER` usage is probably not consistent enough so I’d say ship `TEXT_MAIN` and I can clean up anything that gives me tension in a UI polish PR, never.

**Pull request merged**
